content/blog/clickhouse-benchmarking.md

```diff
@@ -0,0 +1,171 @@
+
+By examining performance metrics during the ingestion of 1 million, 5 million, 10 million, and 66 million logs, we aim to provide a quantitative analysis of how system behavior changes as the load increases.
+
+<p align="left">
+  <img src="/images/blog/clickhouse-benchmarking/clickhouse-cluster-ingestion.png" alt="clickhouse-cluster-ingestion" style="border-radius: 10px; width: 650px; height: 300px;">
+</p>
+
+For more details on the schema and test queries used, you can [download the SQL file here](/images/blog/clickhouse-benchmarking/clickhouse_benchmark_queries.sql).
+
+## Performance Comparison of Key Metrics Across Ingestion Volumes
+
+| `Logs Ingested` | `CPU Usage (Cores)` | `Selected Bytes per Second (B/s)` | `IO Wait (s)` | `CPU Wait (s)` | `Read from Disk (B)` | `Read from Filesystem (B)` | `Memory Tracked (Bytes)` | `Selected Rows per Second` | `Inserted Rows per Second` |
+|-------------------|-----------------------|------------------------------------|-----------------|-----------------|------------------------|------------------------------|----------------------------|------------------------------|------------------------------|
+| `1 Million (10 Lakh)` | 0.103 | 27,118.35 | 0.000 | 0.0217 | 0.000 | 439,023.78 | 714,778,747.45 | 256.88 | 238,666.67 |
+| `5 Million (50 Lakh)` | 0.111 | 37,545.82 | 0.000 | 0.0058 | 0.000 | 532,938.08 | 738,249,399.13 | 256.88 | 238,666.67 |
+| `10 Million (1 Crore)` | 0.433 | 36,265,260.35 | 0.000 | 0.3297 | 1,843.20 | 3,323,855.43 | 1,030,697,939.98 | 146,848.98 | 140,193.58 |
+| `66 Million (6.6 Crore)` | 1.179 | 122,710,173.00 | 0.750 | 4.4276 | 17,330,176.00 | 14,303,693.00 | 2,065,638,032.00 | 365,589.00 | 365,589.00 |
+
+<p align="center">
+  <img src="/images/blog/clickhouse-benchmarking/clickhouse-write-operations.png" alt="clickhouse-benchmarking" style="border-radius: 10px; width: 300; height: 500;">
+</p>
+
+### Key Insights from the Data
+
+#### 1. `CPU Usage (avg_cpu_usage_cores)`
+
+- At 1 million logs, the CPU usage was minimal at `0.103` cores, indicating a low load on the system.
+- This increased marginally to `0.111 cores` when ingesting 5 million logs.
+- However, there was a significant jump when ingesting `10 million logs`, where CPU usage hit `0.433 cores` due to increased processing demands.
+- At `66 million logs`, the CPU usage surged to `1.179 cores`, showing that ClickHouse required far more processing power as the data ingestion rate scaled up.
+
+#### 2. `Selected Bytes per Second (avg_selected_bytes_per_second)`
+
+- `For 1 million logs`, the system processed `27,118 bytes/sec`, and this grew to `37,546 bytes/sec` for `5 million logs`.
```

Comment:
I don't see much insight here. These are just values you already have shown through the image.

Insights are easy to understand and are easily consumable versions of what data shows/infers. We should try to present that.

For example, a `6x` increase in ingestion rate results in a `30%` increase in CPU.